x/coin.go

```diff
@@ -76,7 +76,11 @@ func (c Coin) Divide(pieces int64) (Coin, Coin, error) {
 }
 
 // Multiply returns the result of a coin value multiplication.
-func (c Coin) Multiply(times int64) Coin {
+func (c Coin) Multiply(times int64) (Coin, error) {
```

Comment:
If we enforce int32, I think we guarantee to have no overflows ever....
But we don't have to do so